src/libasr/pass/intrinsic_array_function_registry.h

```diff
@@ -1290,7 +1290,9 @@ namespace Cshift {
         args.push_back(al, result);
         ASR::expr_t *i = declare("i", int32, Local);
         ASR::expr_t *j = declare("j", int32, Local);
-        ASR::expr_t* shift_val = args[1];
+        ASR::expr_t* shift_val = declare("shift_val", int32, Local);;
+        body.push_back(al, b.Assignment(shift_val, b.i2i_t(args[1], int32)));
```

Comment:
```suggestion
        body.push_back(al, b.Assignment(shift_val, args[1]));
```
Why do we need `b.i2i`. I think this will work, no?